From 5b26ee102753d3e526e85e603831c7916ec38750 Mon Sep 17 00:00:00 2001
From: Adriana Kobylak <anoo@us.ibm.com>
Date: Mon, 8 Oct 2018 13:14:12 -0500
Subject: [PATCH] Add new Software Version Incompatible error

Add a new error for when a system component does not meet the
version requirements as determined by the implementation.
Collect the following metadata:
- MIN_VERSION: The minimum version level as defined by the implementation.
- ACTUAL_VERSION: The version running on the system component.
- VERSION_PURPOSE: The Version Purpose as described in
    https://github.com/openbmc/phosphor-dbus-interfaces/blob/master/xyz/openbmc_project/Software/Version.interface.yaml
    or some other descriptive text.

Change-Id: I3e39bfa0735173aee7f3c2177ae24afa619efd69
Signed-off-by: Adriana Kobylak <anoo@us.ibm.com>
---
 xyz/openbmc_project/Software/Version.errors.yaml   | 7 +++++++
 xyz/openbmc_project/Software/Version.metadata.yaml | 9 +++++++++
 2 files changed, 16 insertions(+)
 create mode 100644 xyz/openbmc_project/Software/Version.errors.yaml
 create mode 100644 xyz/openbmc_project/Software/Version.metadata.yaml

diff --git a/xyz/openbmc_project/Software/Version.errors.yaml b/xyz/openbmc_project/Software/Version.errors.yaml
new file mode 100644
index 0000000..4d63a57
--- /dev/null
+++ b/xyz/openbmc_project/Software/Version.errors.yaml
@@ -0,0 +1,7 @@
+- name: Incompatible
+  description: >
+    A system component has a software version that is incompatible as determined
+    by the implementation and needs to be updated. Some usage examples for this
+    error include creating logging events and providing information on
+    implementation reactions such as when the system is prevented from powering
+    on if a minimum version level is not met.
diff --git a/xyz/openbmc_project/Software/Version.metadata.yaml b/xyz/openbmc_project/Software/Version.metadata.yaml
new file mode 100644
index 0000000..155e91d
--- /dev/null
+++ b/xyz/openbmc_project/Software/Version.metadata.yaml
@@ -0,0 +1,9 @@
+- name: Incompatible
+  level: ERR
+  meta:
+    - str: "MIN_VERSION=%s"
+      type: string
+    - str: "ACTUAL_VERSION=%s"
+      type: string
+    - str: "VERSION_PURPOSE=%s"
+      type: string
-- 
1.8.3.1

